MIGRATION.md

```diff
@@ -537,3 +556,646 @@ import { Grid, GridItem } from '@contentful/f36-components';
+  helpText="Some help text"
+  validationMessage="Validation Message"
+/>
+```
+
+will become:
+
+```tsx static=true
+// Checkbox
+<FormControl id="some-id">
+  <Checkbox helpText="Some help text">Your label text</Checkbox>
+  <FormControl.ValidationMessage>validation message</FormControl.ValidationMessage>
+</FormControl>
+
+// TextInput
+<FormControl id="inputId">
+  <FormControl.Label>Counting characters and HelpText</FormControl.Label>
+  <TextInput name="email" maxLength={10} />
+  <Flex justifyContent="space-between">
+    <FormControl.HelpText>Some help text</FormControl.HelpText>
+    <FormControl.Counter />
+  </Flex>
+  <FormControl.ValidationMessage>Validation Message</FormControl.ValidationMessage>
+</FormControl>
+```
+
+For more detailed information and examples, check our documentation website.
+
+#### How to migrate your Field components
+
+Run the [codemod](https://github.com/contentful/forma-36/tree/forma-v4/packages/forma-36-codemod) to migrate your v3 `Field components to the new version:
+
+`npx @contentful/f36-codemod`
+
+The codemods that need to be ran are: `v4-checkbox`, `v4-radio`, `v4-select`, and `v4-text-field`.
+
+If you decide to do it manually, you would need to transform your code:
+
+```tsx static=true
+import { CheckboxField, TextField } from '@contentful/forma-36-react-components';
+
+<CheckboxField
+  labelText="Your label text"
+  id="some-id"
+  type="checkbox"
+  helpText="Some help text"
+  validationMessage="validation message"
+/>
+
+// TextInput
+<TextField
+  id="inputId"
+  name="email"
+  labelText="Counting characters and HelpText"
+  textInputProps={{
+    maxLength: 10,
+  }}
+  helpText="Some help text"
+  validationMessage="Validation Message"
+/>
+```
+
+into this new version:
+
+```tsx static=true
+import { Checkbox, TextInput, FormControl } from '@contentful/f36-components';
+
+<FormControl id="some id">
+  <Checkbox helpText="Some help text">Your label text</Checkbox>
+  <FormControl.ValidationMessage>validation message</FormControl.ValidationMessage>
+</FormControl>
+
+<FormControl id="inputId">
+  <FormControl.Label>Counting characters and HelpText</FormControl.Label>
+  <TextInput name="email" maxLength={10} />
+  <Flex justifyContent="space-between">
+    <FormControl.HelpText>Some help text</FormControl.HelpText>
+    <FormControl.Counter />
+  </Flex>
+  <FormControl.ValidationMessage>Validation Message</FormControl.ValidationMessage>
+</FormControl>
+```
+
+### Checkbox
+
+With the removal of the `CheckboxField` component, we updated how this component works on the version 4, on the version 3 it only rendered the checkbox input without anything else, now it also handles the label and help text for the checkbox. For example:
+
+```jsx static=true
+import { Checkbox } from '@contentful/forma-36-react-components';
+
+<Checkbox id="checkbox" name="checkbox" labelText="some label" />;
+```
+
+This example would render only the input and the `labelText` would be set as the `aria-label`.
```

Comment:
Maybe something like:
Note: This example would render only the input and the `labelText` would be set as the `aria-label`.